### ISSUE TITLE
Formatting buttons not affect all selected text unless a scroll occurs

### ISSUE THREAD

**Issue by fuffalist** [quilljs/quill#1102]
Formatting buttons (B, I, U) are not always formatting all the words in the selected text. Sometimes only the first selected word is formatted. After a scroll, also the other selected words are formatted.

**Steps for Reproduction**

1. Using a tablet (Chrome on Android or Safari on iPad), visit quilljs.com
2. Select a word in the text (for example, the first 'Quill' word)
3. Scroll down to see the standard Quill toolbar at the top of the text area
4. Drag the right selection boundary to select more words
5. DO NOT SCROLL
6. Click on any formatting button in the toolbar (for example: B for Bold)
7. Only the first word of the selected text becomes bold (IMHO this is wrong, all the selected words should become bold)
8. Scroll a little bit up or down
9. Click on any formatting button in the toolbar (example: B for Bold)
10. All the words in the selected text become bold (this is correct)

This issue occurs also when dragging the left boundary or selecting more lines of text.

It seems that a scroll fix the text selection.

**Expected behavior**:
At step 7, all selected words should change their formatting

**Actual behavior**:
At step 7, only the first word becomes formatted

**Platforms**:
Chrome on Android 6.0 tablet (Samsung tab S2)
Safari on iOS 10 over iPad

**Version**:
Android 6.0
iOS 10

Run `Quill.version` to find out
1.1.4

**Comment by jhchen** [comment]
I'm not able to reproduce on iOS 10 Safari on iPad. The selection updates within 100ms of a `touchend` or `touchleave` event so so in theory this could happen if you tapped the format button very quickly or if touch events were being suppressed but the quilljs.com homepage does not do this.

**Comment by fuffalist** [comment]
To reproduce the issue, tap to select one single word, then release the finger and drag the selection margins to cover more words. If you don't scroll and tap on a toolbar button, not all the words in the selected text are formatted.

**Comment by fuffalist** [comment]
I've recorded a short video that highlights this issue and how to reproduce it.
[https://1drv.ms/v/s!Aj5Gl1eygSq0gYoAFKlKWDjSyz1f-A](https://1drv.ms/v/s!Aj5Gl1eygSq0gYoAFKlKWDjSyz1f-A)

**Comment by jhchen** [comment]
Is this still an issue? I reproduced it and tagged it as a bug in Dec but can no longer reproduce.

**Comment by fuffalist** [comment]
Hi,

I still see this issue. I've just reproduced it with Chrome on a Galaxy S6. Here are the steps:

**Steps for Reproduction**

1. Login to quilljs.com onto a mobile device
1. Switch to the first example (first dot on top of the editor, itthe one with the context menu)
1. Double tap on a word in the editor (I tested with the word WYSIWYG)
1. Tap on the right blue point and drag right to get some words on the right (WYSIWYG editor build for the ...)
1. Tap on the B button
1. You'll see that only the first word (WYSIWYG in my example) appear in bold

**Expected behavior:**
At step 6, all selected words should change their formatting

**Actual behavior:**
At step 6, only the first word is formatted

**Tetmporary workaround:**
Between steps 4 and 5, scroll up or down a little bit

**Platforms:**
Chrome on Android 7.0 (Samsung Galaxy S6)
